src/averageAges.js

```diff
@@ -55,7 +79,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    (onlyWithSon ? person.sex === 'm' : true)
+    && people.find(mother => mother.name === person.mother)
+  );
+
+  const ageDifference = children.map(child => {
+    const personMother = people.find(mother => mother.name === child.mother);
+
+    return child.born - personMother.born;
+  });
+
+  const yearsSum = ageDifference.reduce((sum, years) =>
+    sum + years, 0);
+
+  return yearsSum / ageDifference.length;
```

Comment:
This logic is repeated in every function. Let's put this in a separate function `calculateAverageAge` that will take an array of numbers and a total number